lib/streamlit/config_util.py

```diff
@@ -91,40 +89,52 @@ def append_newline():
             key = option.key.split(".")[1]
             description_paragraphs = _clean_paragraphs(option.description)
 
-            for i, txt in enumerate(description_paragraphs):
-                if i == 0:
-                    append_desc("# %s" % txt)
-                else:
-                    append_comment("# %s" % txt)
+            last_paragraph_idx = len(description_paragraphs) - 1
+
+            for i, paragraph in enumerate(description_paragraphs):
+                # Split paragraph into lines
+                lines = paragraph.split("\n")
+
+                # If the first line is empty, remove it
+                if lines and not lines[0].strip():
+                    lines = lines[1:]
+
+                # Choose function based on whether it's the first paragraph or not
+                append_func = append_desc if i == 0 else append_comment
+
+                # Add comment character to each line and add to out
+                for line in lines:
+                    append_func(line.lstrip())
+
+                # Add a line break after a paragraph only if it's not the last paragraph
+                if i != last_paragraph_idx:
+                    out.append("")
```

Comment:
Could we add a test in `config_util_test.py` that calls `show_config` with simple `section_descriptions` / `config_options` arguments to verify that this formatting code works as expected? I think this would be easiest to do by importing the raw `ConfigOption` class from `streamlit.config_option` and creating one directly

It seems to me like we're missing a good amount of test coverage for this file (I'm probably at least partially responsible for this 😅). I wouldn't worry about improving the test coverage overall in this PR (I'm happy to add this to my backlog), but I do think we should add some coverage specifically for this section of code.

Reply:
Yep, this covers my concerns and then some! I'll see if I can find the free time to add more tests in a future PR, but for now these are great.